### ISSUE TITLE
Closing brace hints move away from the cursor when typing

### ISSUE THREAD

**Issue by Wilfred** [rust-lang/rust-analyzer#16636]
Inserting a new item after a closing brace with hint can lead to the hint moving. This is also awkward considering that the hint itself can be selected.

https://github.com/rust-lang/rust-analyzer/assets/70800/58f708be-bd03-4836-84f0-b86d9f7392bd

Tested on rust-analyzer trunk as of 2024-02-16.

**Comment by Veykril** [comment]
That's VSCode not updating inlay hints immediately (they still have an annoying artificial delay employed), so not something we can fix.

**Comment by davidbarsky** [comment]
Marking this as a wontfix as it's blocked by https://github.com/microsoft/vscode/issues/145377.
